src/lib.rs

```diff
@@ -6,12 +6,39 @@
 
 use once_cell::sync::Lazy;
 
+#[cfg(feature = "kv")]
+use slog::KV;
 use tracing_core::{
     callsite, dispatcher, field, identify_callsite,
     metadata::{Kind, Level},
     subscriber, Event, Metadata,
 };
 
+#[cfg(feature = "kv")]
+/// An allocating serializer to use for serializing key-value pairs in a [`slog::Record`]
+#[derive(Default)]
+struct TracingKvSerializer {
+    storage: String,
+}
+
+#[cfg(feature = "kv")]
+impl TracingKvSerializer {
+    fn as_str(&self) -> &str {
+        self.storage
+            .get(..self.storage.len().saturating_sub(1))
+            .unwrap_or_default()
```

Comment:
I was wondering how to make this simpler, but I'm not sure if this is: 
```suggestion
        &self.storage[..1.max(self.storage.len())-1]
```

Reply:
This was a bit simplified in the newest commit and makes better use of the `Option` we get from `get()` vs direct slice access